pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -102,6 +103,12 @@ public void init(TableDataManagerConfig tableDataManagerConfig, String instanceI
     if (!_indexDir.exists()) {
       Preconditions.checkState(_indexDir.mkdirs());
     }
+    _resourceTmpDir = new File(_indexDir, "tmp");
+    // Delete and recreate the tmp directory if it exists as it may contain old files from previous runs.
+    _resourceTmpDir.delete();
```

Comment:
Could we use `FileUtils.deleteQuietly`, also is the `exists` check needed? I take it that the `exists()` check is needed in case the delete failed to delete existing directory?

Reply:
+1 on deleting the directory quietly. We don't want an exception thrown during the startup of the server.

Reply:
good point, done. the exists check really shouldn't be needed here, but I figure it can't hurt since we're using `Preconditions` afterwards